### ISSUE TITLE
Docs: [prefer-nullish-coalescing] doesn't work with `strictNullChecks: false`.

### ISSUE THREAD

**Issue by jsalvata** [typescript-eslint/typescript-eslint#5748]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.8.4&sourceType=module&code=DYUwLgBAHgXBDOYBOBLAdgcwgHwgVzQBMQAzdEQgbgCgodcByBmoA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WlqYSNkAC1pkA9gEMkyMswDm6KL2jjokcGAC+ILUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylAGYBDVCXTgwAXxCygA

### Repro Code

```typescript
let x: string | undefined;
x || '';
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  "rules": {
    "@typescript-eslint/prefer-nullish-coalescing": "error"
  }
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": false
  }
}
```

### Expected Result

An error should be reporting for the `||`, which should be replaced with `??`.

### Actual Result

No error is reported.

### Additional Info

The same rule works if `strictNullChecks` is changed to `true`.

This makes sense because `strictNullChecks: false` essentially erases `| null` and `| undefined` from all types.

Ideally the rule should work anyway, assuming all types to be nullable. If that's not possible, a note should be added to the documentation so people do not spend a whole afternoon trying to understand why the rule doesn't work, as I did.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.39.0` |
| `@typescript-eslint/parser`            | `5.39.0` |
| `TypeScript`                           | `4.4.4` |
| `ESLint`                               | `7.32.0` |
| `node`                                 | `12.22.6` |

**Comment by bradzacher** [comment]
This is expected.
With the compiler option turned off TS essentially erases the nullish types from everything. So when we inspect the types - they don't include the nullish types.

Use the flag - it's a VERY bad practice to not use it.

**Comment by jsalvata** [comment]
Even if working as intended, which I agrree with, a note should be added to the documentation for the sake of the rule user's.

Bad practice indeed, but the flag is there for a reason: sometimes it is unavoidable, at least temporarily.

**Comment by JoshuaKGoldberg** [comment]
True! Marking as accepting PRs for a docs update to mention this on https://typescript-eslint.io/rules/prefer-nullish-coalescing.

At least one other rule has a similar limitation. You can probably copy the phrasing from it/them.

**Comment by bradzacher** [comment]
We should give it the same treatment as rules like `strict-boolean-expressions` where the rule will explicitly error if you don't have the option turned on.

https://typescript-eslint.io/rules/strict-boolean-expressions/#allowruletorunwithoutstrictnullchecksiknowwhatiamdoing

**Comment by JoshuaKGoldberg** [comment]
For anybody interested in this as a `good first issue`: feel free to do either or both of:

* Adding an explicit notice to the docs page (`prefer-nullish-coalescing.md`)
* Adding a similar option to the rule

We can file a followup issue if you only tackle one and not the other. 🙂

**Comment by JoshuaKGoldberg** [comment]
Reopening so @cparros can tackle the second part 😄
